server/bin/www

```diff
@@ -87,3 +90,33 @@ function onListening() {
 
   console.log(`Listening on ${bind}`.red.underline.bold);
 }
+
+/**
+ * Authentication and listener for user connection to socket
+ */
+
+io.use(async (socket, next) => {
+  const token = socket.handshake.headers.cookie
+
+  if (!token) {
+    console.log("Invalid token")
+  }
+
+  try {
+    const userToken = token.split("token=").join("");
+
+    const decodedUser = jwt.verify(userToken, process.env.JWT_SECRET);
+
+    const user = await User.findById(decodedUser.id);
+    if (!user) {
+      console.log("User not found");
+    }
+
+    socket.data.user = user
+    next();
+  } catch (err) {
+    console.log("User authentication failed!")
+  }
+}).on("connection", (socket) => {
+  console.log(socket.data.user)
```

Comment:
Let's get rid of console logs

Reply:
The bottom one was explicity required by the ticket (and confirmed), but yes the first one needs converting into an error.